### ISSUE TITLE
Warning printed when using docker_container on devel branch

### ISSUE THREAD

**Issue by lattwood** [ansible/ansible#20521]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
plugins core AND/OR docker_container module

##### ANSIBLE VERSION
SHA: e3c95615b2117ab4a5ebb43f1c96272bc252dbb6

##### SUMMARY
Getting this in the playbook run output when using `docker_container`:
```
 [WARNING]: Removed restricted key from module data: ansible_docker_container = REDACTED
```
Looks like it was introduced by this comment by @bcoca https://github.com/ansible/ansible/commit/1880027da19d223819d83dd84325edc795783648

##### STEPS TO REPRODUCE
Run devel ansible, use docker_container module

##### EXPECTED RESULTS
No warning displays

##### ACTUAL RESULTS
A warning displays

**Comment by lattwood** [comment]
Dug in, here's some more (potential) details, I think.

It's because of the name of the return fact- `ansible_docker_container`: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py#L1675

I think the only way it could be added to remove_keys is via this regex: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/__init__.py#L700

**Comment by bcoca** [comment]
docker container is returning keys that conflict with the docker connection plugin, I need to check and see if we need an exception or to rename the keys

**Comment by lattwood** [comment]
I can confirm the above Quickfix solved the issue.

**Comment by DerekTBrown** [comment]
+1

**Comment by xiemeilong** [comment]
+1

**Comment by xiemeilong** [comment]
+1

**Comment by bcoca** [comment]
fixed via #22324

**Comment by mvasin** [comment]
Here's my task:
```
  - name: Create containers' network.
    docker_network:
      name: ntv_network
```

Here's the output:
```
 _____________________________________________________
< TASK [Create containers' network. name=ntv_network] >
 -----------------------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

 [WARNING]: Removed restricted key from module data: ansible_docker_network = {u'Options': {}, u'Name': u'ntv_network', u'Created': u'2017-05-24T13:09:35.454684783+03:00',
u'EnableIPv6': False, u'Labels': {}, u'Driver': u'bridge', u'Id': u'84d9aa2121429f007399b55e79d1ca82fb81f125bfc51178158db2c16e194f78', u'Internal': False, u'IPAM': {u'Config':
[{u'Subnet': u'172.18.0.0/16', u'Gateway': u'172.18.0.1'}], u'Driver': u'default', u'Options': None}, u'Scope': u'local', u'Attachable': False, u'Containers':
{u'f30256ae1f93d65b12e7529c517bb1634e83e66cc4d356330330a12f02f4742d': {u'IPv4Address': u'172.18.0.2/16', u'MacAddress': u'02:42:ac:12:00:02', u'Name': u'postgres', u'EndpointID':
u'f9f65016ebc9052250cec27bbf3e5acef3723a3b6f62ac6e4d0022db643dd140', u'IPv6Address': u''}}}

ok: [server]
```

**Comment by michael-k** [comment]
@mvasin The network warning is fixed in devel (aka ansible 2.4) and has been cherry picked for an eventual 2.3.1 release.  See ansible/ansible#23919
